include/mapbox/variant_visitor.hpp

```diff
@@ -13,7 +15,7 @@ struct visitor<Fn> : Fn
     using type = Fn;
     using Fn::operator();
 
-    visitor(Fn fn) : Fn(fn) {}
+    visitor(Fn&& fn) : Fn(std::forward<Fn>(fn)) {}
```

Comment:
Thank you @jfirebaugh, you're right!
Your test gives me a `Fn` deduced as a `lambda&`.
I included your case to `test_match_singleton()` and added some more cases to `test_match_overloads_init_capture()`. There is a fix at d4cd55a897f24feb2a2bdb2a9b16cba7a414c0b7.

Can you take a look, please?